packages/agents/src/tests/index.test.ts

```diff
@@ -0,0 +1,672 @@
+        })
+      );
+
+      expect(response).toBeDefined();
+      expect(response.status).toBe(200);
+    });
+  });
+
+  describe("StreamingResponse", () => {
+    it("should require connection and id parameters", () => {
+      expect(StreamingResponse).toBeDefined();
+      expect(typeof StreamingResponse).toBe("function");
+    });
+
+    it("should be used for streaming RPC responses", () => {
+      const mockConnection = {
+        id: "test-connection-id",
+        state: {},
+        setState: vi.fn(),
+        server: "test-server",
+        send: vi.fn()
+      };
+
+      const response = new StreamingResponse(
+        mockConnection as unknown as Connection,
+        "test-id"
+      );
+      expect(response).toBeDefined();
+
+      response.send({ data: "chunk1" });
+      expect(mockConnection.send).toHaveBeenCalledWith(
+        JSON.stringify({
+          done: false,
+          id: "test-id",
+          result: { data: "chunk1" },
+          success: true,
+          type: "rpc"
+        })
+      );
+
+      response.end({ data: "final" });
+      expect(mockConnection.send).toHaveBeenCalledWith(
+        JSON.stringify({
+          done: true,
+          id: "test-id",
+          result: { data: "final" },
+          success: true,
+          type: "rpc"
+        })
+      );
+
+      expect(() => response.send({ data: "after-end" })).toThrow(
+        "StreamingResponse is already closed"
+      );
+    });
+  });
+
+  describe("Agent Class Features", () => {
+    it("should support callable decorator", () => {
+      const metadata = {
+        description: "Test method",
+        streaming: false
+      };
+
+      const decorator = unstable_callable(metadata);
+      expect(typeof decorator).toBe("function");
+    });
+
+    it("should handle different agent namespaces", async () => {
+      const testAgent = await getAgentByName(env.TEST_AGENT, "test-agent");
+      expect(testAgent).toBeDefined();
+      expect(testAgent.fetch).toBeDefined();
+
+      const emailAgent = await getAgentByName(env.EmailAgent, "email-agent");
+      expect(emailAgent).toBeDefined();
+      expect(emailAgent.fetch).toBeDefined();
+
+      const caseSensitiveAgent = await getAgentByName(
+        env.CaseSensitiveAgent,
+        "case-agent"
+      );
+      expect(caseSensitiveAgent).toBeDefined();
+      expect(caseSensitiveAgent.fetch).toBeDefined();
+
+      const notificationAgent = await getAgentByName(
+        env.UserNotificationAgent,
+        "notification-agent"
+      );
+      expect(notificationAgent).toBeDefined();
+      expect(notificationAgent.fetch).toBeDefined();
+    });
+  });
+
+  describe("getCurrentAgent", () => {
+    it("should return context when called", () => {
+      // getCurrentAgent returns the context object, not just the agent
+      const context = getCurrentAgent();
+
+      if (context) {
+        expect(context).toHaveProperty("agent");
+      } else {
+        expect(context).toBeUndefined();
```

Comment:
this doesn't test anything, need more here. probably all scenarios: read from getCurrentAgent in well known methods, as well as custom methods, and assert that agent is defined and points to the object. also test request/connection/email